### DISCUSSION TITLE
Is there a way to return documents from one's own onedrive only in PnP Modern Search?

### DISCUSSION THREAD

**Discussion by mw-new-test** [microsoft-search/pnp-modern-search#3080] [Q&A] [answered]
We are currently fetching results from OneDrive in the following way.
{searchTerms}* AND (SPSiteUrl: https://mycompany-my.sharepoint.com/personal/*) AND IsDocument:1

If there are some confidential documents on any user's OneDrive but accidentally shared with Everyone in the organization, those documents show up in the results with the above Query template.

Is there a way to restrict search results to return files from user's own onedrive?
Ideally we would like to get only the files that are created/modified by the user on their own onedrive or modified by the user on someone else's onedrive.

Also, is it possible to include a handlebar helper within the Query template?
Typically the Onedrive URL is something like --> https://mycompany-my.sharepoint.com/personal/abc_xyz_mycompany_com
We would like to pass the user's email ID in the Query template by replacing the '.','-'and '@' with '_'
We tried the below query but it does not work.
{searchTerms}* AND (SPSiteUrl: https://mycompany-my.sharepoint.com/personal/{{replace User.Email (regex '.-@' (User.Email)) '_'}}*) AND IsDocument:1

Any help here is appreciated.

**Comment by kasperbolarsen** [comment, marked as answer]
You should be able to limit the search to only onedrive by looking at the SiteTemplate property, I haven't tested it but I think it is: SiteTemplate:spspers*

**Reply by mw-new-test** [reply]
Thanks @kasperbolarsen, I am able to restrict onedrive results with - SPSiteUrl: https://***-my.sharepoint.com/personal/* same as SiteTemplate:spspers*.
The problem is that it returns driveitems from my onedrive and also files that are shared from every user's onedrive.
I would like users to see their own onedrive data and not data that exists on other people's onedrive which may be confidential and accidentally shared with everyone.

**Reply by mw-new-test** [reply]
Like how there is a Graph API to get current user's drive items - /me/drive,
we are looking for a similar way to check if there a way in KQL to restrict results to logged-in user's onedrive items.

**Comment by kasperbolarsen** [comment]
Tricky one. Perhaps the user property "PersonalSpace" can be of use by combining it with the {User.<property>} . So perhaps something like Path:{User.PersonalSpace} ? 
However, it looks like the PersonalSpace property is not indexed by default, so you might have to set it to be indexed, and run one of the PowerShell scripts that forces a "full index" ?

**Comment by kasperbolarsen** [comment]
Using PersonalSpaces ( after making it indexed, and mapping it to a new MP) seems to work:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/426d9360-0456-405a-af73-5535f3ef8ce1)

**Reply by mw-new-test** [reply]
Thanks @kasperbolarsen for investigating.
We had some restrictions to index PersonalSpace user profile property.
We have currently implemented using the below query.

{searchTerms}* AND (SPSiteUrl: https://mycompany-my.sharepoint.com/personal/*) AND (AuthorOWSUSER:{User.Email}* OR EditorOWSUSER:{User.Email}* OR SharedWithUsersOWSUSER:{User.Email}*) AND IsDocument:1 

This returns documents that were created by me, modified by me and shared individually with me which is fine. However, there is one drawback which is - when a document is shared with a group that I am part of, that document is not returned in the results.

But atleast now we do not see the many confidential documents that users accidentally shared.
